Implement PagingPredicate (#232)
hazelcast/hazelcast-python-client: README.md

Also, you can sort the entries by specifying comparators. In this case, the comparator should be either `Portable` or `IdentifiedDataSerializable` 
and the serialization factory implementations should be registered on the member side. Please note that,
paging is done on the cluster members. Hence, client only sends a marker comparator to indicate members
which comparator to use. The comparision logic must be defined on the member side by implementing the 
`java.util.Comparator<Map.Entry>` interface.

Paging predicates require the objects to be deserialized on the member side from which the collection is retrieved. 
Therefore, you need to register the serialization factories you use on all the members on which the paging predicates are used.
See the [Adding User Library to CLASSPATH](#1212-adding-user-library-to-classpath) section for more details.

In the example code below:

- The `is_greater_than_or_equal_to` predicate gets values from the `students` map. 
This predicate has a filter to retrieve the objects with an `age` greater than or equal to `18`.

- Then a `PagingPredicate` is constructed in which the page size is `5`, so that there are five objects in each page. 

> The first time the values are called creates the first page.

Reviewer: nit: I'm not sure I understand what's meant by this sentence. It's better to re-write it in a clearer way.

Author: I think it is related to the ``values`` method used in the code sample, rewrite that part

> - It gets subsequent pages with the `next_page()` method of `PagingPredicate` and querying the map again with the updated `PagingPredicate`.

Reviewer: ```suggestion
- Finally, the subsequent page is fetched by calling the `next_page()` method of `PagingPredicate` and querying the map again with the updated `PagingPredicate`.
```

Author: Changed

```python
from hazelcast.serialization.predicate import paging, is_greater_than_or_equal_to

...

m = client.get_map("students").blocking()
predicate = paging(is_greater_than_or_equal_to("age", 18), 5)

# Retrieve the first page
values = m.values(predicate)

...

# Set up next page
predicate.next_page()

# Retrieve next page
values = m.values(predicate)
```

> If a comparator is not specified for `PagingPredicate`, but you want to get a collection of keys or values page by page, 

Reviewer: nit: does it make sense to stress the fact that the `If a comparator is not specified for `PagingPredicate`` part speaks of the cluster-side class? Python users may be confused otherwise.

Author: I believe this part is meant for the cases like `paging(predicate, 5, None)` or `paging(predicate, 5)` when the comparator is not specified on the client-side. Did I get your question wrong?

Reviewer: Probably, I misread this part. Thanks for the clarification. Resolving this one.

> this collection must be an instance of `Comparable` (i.e., it must implement `java.lang.Comparable` on the member side). 

Reviewer: nit: maybe it's more precise to say that keys/values should implement `Comparable`, not their collections?

Author: Good idea, changed that part

Otherwise, paging fails with an exception from the server. Luckily, a lot of types implement the `Comparable`
interface by [default](https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html), including the primitive types,
so, you may use values of types `int`, `float`, `str` etc. in paging without specifying a comparator on the Python client.

You can also access a specific page more easily by setting the `predicate.page` attribute before making the remote call.
This way, if you make a query for the hundredth page, for example, it gets all `100` pages at once instead of reaching the hundredth page one by one using the `next_page()` method. 

> `PagingPredicate`, also known as Order & Limit, is not supported in Transactional Context.

Reviewer: nit: does it make sense to convert this one into a note (smth like `> **NOTE: foo bar**` style used in Node.js client)? It seems to be a side note which falls of the previous context.

Author: Added it
